tutorial/index.md

```diff
@@ -79,48 +82,38 @@ juju offer self-signed-certificates:send-ca-cert
+# create the user and get the identity-id
+juju run kratos/0 create-admin-account email=test@example.com password=test username=admin
+# reset the password to make it active
+juju add-secret password-secret password=abc
+juju grant-secret password-secret kratos
+juju run kratos/0 reset-password identity-id=<identity-id> password-secret-id=<secret:id>
 ```
 
-Setting up an IdP to point to your local environment can be tricky depending on the provider used. Below are some tips to make this work.
-
-When setting up certain providers, e.g. Google, security restrictions limit what redirect URLs can be used. Some restrictions include,
-
-- The redirect URL must be `https`.
-- The redirect URL must be a top level domain `.com`.
-- The redirect URL cannot be an IP address.
-
-The redirect URL is the URL that your browser is returned to after you have signed in at the identity provider. When using Canonical's
-identity bundle, the redirect URL after login will be something like `https://<kratos-public-url>/self-service/methods/oidc/callback/<provider-id>`.
-as described in the above how-to. Although the URL is `https`, it is an IP address.
-This address needs to be registered in your identity provider as an approved redirect URI/URL.
-
-If your preferred identity provider does not accept an IP address, we recommend using a tool like `https://nip.io/`,
-a DNS resolver service that can map any IP address to a hostname.
-
-| This service can map hostnames of the form `<anything>[.-]<IP Address>.nip.io` to return simply `<IP Address>`.
-| E.g. `magic.127.0.0.1.nip.io` resolves to `127.0.0.1`. This service is very useful when working with an IdP locally for testing.
+### Expose the identity bundle to your host machine.
+The reason you need to expose the identity bundle to your host machine is that at the end of this tutorial
+you will need to login via a web browser.
 
-```{note}
-The same effect can be obtained by editing your `/etc/hosts` file but this would require changes on your host system and within various containers.
+```text
+# locate the ip of your multipass instance. It should be the ip for second interface.
+# you can double check it by doing `multipass list` in your host machine.
+ip addr
+juju config traefik-public external_hostname=<multipass-ip>
+# port-forward traefik pod to expose the service outside of the multipass instance
+sudo microk8s.kubectl port-forward traefik-public-0 443:443 --namespace=iam --address=<multipass_ip> &
 ```
 
-To utilise `nip.io`, get the address of your `traefik-public` instance and set the `external_hostname` config option as below,
-
+To test you've managed to expose the service you can
 ```text
-TRAEFIK_PUBLIC=$(juju status traefik-public --format yaml | yq .applications.traefik-public.address)
-juju config traefik-public external_hostname="iam.$TRAEFIK_PUBLIC.nip.io"
+# in your host machine
+curl -k https://<multipass-ip>/iam-hydra/health/ready
+# the response should be {"status":"ok"}
```

Comment:
This is another example where I think it's confusing, the sentence above runs on into the code block. Just
```suggestion
Run the following on your host machine to test that you've successfully exposed:
$ curl -k https://<multipass-ip>/iam-hydra/health/ready
{"status":"ok"}
```
I'm not quite sure about the `$` but hopefully you get my point.